### ISSUE TITLE
 node modules paths are being resolved to "matching" user workspace paths causing broken package loading

### ISSUE THREAD

**Issue by thescientist13** [ProjectEvergreen/greenwood#686]
## Type of Change
- [ ] New Feature Request
- [ ] Documentation / Website
- [ ] Improvement / Suggestion
- [x] Bug
- [ ] Other (please clarify below)

## Summary
After upgrading a handful of projects to Greenwood v0.15.0, the SPA projects, which also happen to be on Lit@2 RC versions, started to experience this issue _just_ when running `greenwood develop`.  Running `greenwood serve` was fine.
![Screen Shot 2021-08-06 at 12 03 30 PM](https://user-images.githubusercontent.com/895923/128543096-31648cad-1525-497b-92c9-c8ea944768e2.png)

Seems like a regression then if they were all working fine before?  Should try and solve ASAP though along with keeping an eye on #675.

## Details
Examples of the issue
- https://github.com/AnalogStudiosRI/www.analogstudios.net/pull/16
- https://github.com/ContributaryCommunity/www.contributary.community/pull/94

Other projects, (those not using **Lit@2** anyway), seem to be fine with the upgrade, so there's that going for us at least 😅 
- https://github.com/thegreenhouseio/www.thegreenhouse.io/pull/235
- https://github.com/ProjectEvergreen/greenwood-getting-started/pull/43
- https://github.com/ProjectEvergreen/projectevergreen.github.io/pull/60

**Comment by thescientist13** [comment]
OK, found the issue and this is actually related to the refactoring for #520 and more of a coincidence than anything wrong with Lit2@ or SPA mode, per se.

What was happening was in the SPA repo, we had this file structure
```
src/
  index.js
```

In _index.js_ we would `import` Lit as usual
```js
import { html, LitElement } from 'lit';
```

But what I observed in the browser was that Greenwood was actually resolving _node_modules/lit/index.js_ to _src/index.js_!  😱 
![Screen Shot 2021-08-06 at 2 15 21 PM](https://user-images.githubusercontent.com/895923/128559080-96065d24-26d4-4dc8-bc67-2aead2c81cd1.png)

What was happening was that in [plugin-user-workspace](https://github.com/ProjectEvergreen/greenwood/blob/v0.15.0/packages/cli/src/plugins/resource/plugin-user-workspace.js#L21), the [`resolveRelativeUrl`method from `ResourceInterface`](https://github.com/ProjectEvergreen/greenwood/blob/v0.15.0/packages/cli/src/lib/resource-interface.js#L20) was chopping down the path such once it cutoff _node_modules_ and _lit_, it would only leave _index.js_, which we of course have as a file in our _src/_ directory, and thus, a (undesired) "match".
```sh
resolve nodeModulesUrl @ /Users/owenbuckley/Workspace/analogstudios.net/repo/www.analogstudios.net/node_modules/bootstrap/dist/css/bootstrap.css.map
pshouldServe!!! /Users/owenbuckley/Workspace/analogstudios.net/repo/www.analogstudios.net/node_modules/bootstrap/dist/css/bootstrap.css.map
shouldResolve nodeModulesUrl @ /node_modules/lit/index.js
resolve nodeModulesUrl @ /Users/owenbuckley/Workspace/analogstudios.net/repo/www.analogstudios.net/node_modules/lit/index.js
!workspace URL start! @@@@@@@@@@@ /node_modules/lit/index.js
!workspace URL! @@@@@@@@@@@ /Users/owenbuckley/Workspace/analogstudios.net/repo/www.analogstudios.net/src/index.js
pshouldServe!!! /Users/owenbuckley/Workspace/analogstudios.net/repo/www.analogstudios.net/src/index.js
```

--- 

So basically, easiest solution is to just make plugin-user-workspace ignore anything with _node_modules_ in the URL since it doesn't need to care about anything but files in the user's workspace.

Should find a way to add a spec for this when opening the PR.
